src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDcollector.java

```diff
@@ -0,0 +1,208 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions
+ * and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions
+ * and the following disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+ * or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+package org.magicdgs.thaplv.tools.ibd.engine;
+
+import org.magicdgs.thaplv.io.FastaNsCounter;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFHeader;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Collector for variants to compute IBD.
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class IBDcollector {
+
+    // logger for this class
+    private final static Logger logger = LogManager.getLogger(IBDcollector.class);
+
+    // queue for store difference windows until they are not overlapping
+    private final Deque<PairwiseDifferencesWindow> queue;
+
+    // the name of the samples
+    private final List<String> sampleNames;
+
+    // a simple class for count the number of Ns in a sequence
+    protected final FastaNsCounter nCounter;
+
+    // the window size
+    private final int windowSize;
+
+    // the window step
+    private final int windowStep;
+
+    // cached maximum number of windows in RAM
+    private final int maximumWindow;
+
+    /**
+     * New IBD collector for the samples in a VCF header
+     *
+     * @param header     the header where the samples are stored
+     * @param nCounter   the class for count the Ns in the reference
+     * @param windowSize the window size
+     * @param windowStep the window step
+     */
+    public IBDcollector(final VCFHeader header, final FastaNsCounter nCounter, final int windowSize,
```

Comment:
Add a ticket for include only some samples using the GATK command line parameters in the tools and modifying the collector depending on how this is implemented in their framework.